fendermint/vm/interpreter/src/fvm/gas/actor.rs

```diff
@@ -96,13 +152,18 @@ impl ActorGasMarket {
             gas_premium: Default::default(),
         };
 
+        self.call_fvm(msg, executor)?;
+        Ok(())
+    }
+
+    fn call_fvm<E: Executor>(&self, msg: FvmMessage, executor: &mut E) -> anyhow::Result<ApplyRet> {
         let raw_length = fvm_ipld_encoding::to_vec(&msg).map(|bz| bz.len())?;
         let apply_ret = executor.execute_message(msg, ApplyKind::Implicit, raw_length)?;
 
         if let Some(err) = apply_ret.failure_info {
             anyhow::bail!("failed to update EIP1559 gas state: {}", err)
         } else {
-            Ok(())
+            Ok(apply_ret)
         }
     }
```

Comment:
Can be simplified (GitHub won't let me make a suggestion):
 
```rust
apply_ret.failure_info
    .map_or(Ok(apply_ret), |err| anyhow::bail!("failed to update EIP1559 gas state: {}", err))
```

Reply:
I tried, seems the above does not compile due to partial move because `map_or` takes self which consumes `apply_ret.failure_info` and Ok(apply_ret) is partially moved. I tried the same with `as_ref`, but have the same issue. It works if I do `apply_ret.failure_info.take`, but perhaps not desired as it changes the failure info.